src/client.js

```diff
@@ -104,17 +105,34 @@ class Client {
   /**
    * Resolve the DNS shortname `name`. If `name` is already a cabal key,  it will
    * be returned and the DNS lookup is aborted.
+   * If `name` is a whisper:// key, a DHT lookup for the passed-in key will occur. 
+   * Once a match is found, it is assumed to be a cabal key, which is returned.
    * Returns the cabal key in `cb`. If `cb` is null a Promise is returned.
-   * @param {string} name the DNS shortname
-   * @param {function(string)} [cb] The callback to be called when DNS lookup succeeds
+   * @param {string} name the DNS shortname, or whisper:// shortname
+   * @param {function(string)} [cb] The callback to be called when lookup succeeds
    */
   resolveName (name, cb) {
-    return this.cabalDns.resolveName(name).then((key) => {
-      if (key === null) return null
-      key = Client.scrubKey(key)
-      if (!cb) return key
-      else cb(key)
-    })
+    if (name.startsWith("whisper://")) {
```

Comment:
Curious about UX here -- if we're trying to remove barriers for in-person (non copy/paste), is the idea that someone will type also `whisper://` and then the whisper string? Or is this being constructed from the user input?

We can expect five input scenarios:

1. `cabal://HEX32` URI plus 32 char hex string (copy-paste from ui)
2. `HEX32` a 32 char hex string (lazy copy-paste from ui, custom client, testing)
3. `http[s]://[abc..]` a DNS shortname
4. `whisper://[abc...]` a whisper link
5. `[abc...]` a DNS shortname OR a whisper link